src/lib/api/common/client/commonUtils.client.ts

```diff
@@ -24,25 +27,31 @@ export const request = async <T>({
   method: HttpMethod;
   options?: Omit<RequestInit, 'method' | 'headers'>;
 }): Promise<T> => {
+  const requestHeaders = new Headers();
   const token = await getToken();
-  if (!token) {
+
+  /**
+   * Set the Authorization token in the header
+   * If the API is a mock API, then skip this if statement.
+   */
+  if (!isMockApi() && !token) {
     throw new Error('Authentication token is not available');
+  } else if (token) {
```

Comment:
I think this `else` clause is not needed because token definitely exists.

Reply:
The reason why I added this `else if` clause is to skip setting the authorization header if `isMock()` is `true`.

Maybe the code below looks simpler. I tried to avoid nested `if` statements, but in this case, nesting them increases readability. What do you think?

```ts
  if (!isMockApi()) {
    if (!token) throw new Error('Authentication token is not available');
    requestHeaders.set('Authorization', `Bearer ${token}`);
  }
```